flink-staging/flink-ml/src/main/scala/org/apache/flink/ml/math/SparseVector.scala

```diff
@@ -85,6 +85,34 @@ case class SparseVector(
     }
   }
 
+  /** Returns the outer product (a.k.a. Kronecker product) of `this`
+    * with `other`. The result is given in [[org.apache.flink.ml.math.SparseMatrix]]
+    * representation.
+    *
+    * @param other a Vector
+    * @return the [[org.apache.flink.ml.math.SparseMatrix]] which equals the outer product of `this`
+    *         with `other.`
+    */
+  override def outer(other: Vector): SparseMatrix = {
+    val numRows = size
+    val numCols = other.size
+
+    val otherIndices = other match {
```

Comment:
it should be enough to write `otherIndices: Iterable[Int]` and then remove the `toArray` method from the `Range` object.